src/main/java/crawlercommons/domains/PaidLevelDomain.java

```diff
@@ -64,60 +48,20 @@ public static String getPLD(String hostname) {
             return hostname;
         }
 
-        String[] subNames = hostname.split("\\.");
-        int numPieces = subNames.length;
-        if (numPieces <= 2) {
-            return hostname;
-        }
-
         // Check for ddd.ddd.ddd.ddd IPv4 format
-        if ((numPieces == 4) && (IPV4_ADDRESS_PATTERN.matcher(hostname).matches())) {
+        if (IPV4_ADDRESS_PATTERN.matcher(hostname).matches()) {
             return hostname;
         }
 
-        int firstHostPiece = 0;
-        if (ccTLDs.contains(subNames[numPieces - 1].toLowerCase(Locale.ROOT))) {
-            // We have a country code at the end. See if the preceding piece is
-            // either
-            // a two-letter name (country code or funky short gTLD), or one of
-            // the
-            // "well-known" gTLDs.
-            if (subNames[numPieces - 2].length() <= 2) {
-                // Must be xxx.co.jp format
-                firstHostPiece = numPieces - 3;
-            } else if (gTLDs.contains(subNames[numPieces - 2].toLowerCase(Locale.ROOT))) {
-                // Must be xxx.com.mx format
-                firstHostPiece = numPieces - 3;
-            } else {
-                // Must be xxx.it format
-                firstHostPiece = numPieces - 2;
-            }
-        } else if (gTLDs.contains(subNames[numPieces - 1].toLowerCase(Locale.ROOT))) {
-            if (ccTLDs.contains(subNames[numPieces - 2].toLowerCase(Locale.ROOT))) {
-                // Must be xxx.de.com format
-                firstHostPiece = numPieces - 3;
-            } else {
-                // Must be xxx.com format
-                firstHostPiece = numPieces - 2;
-            }
+        // Now use support in EffectiveTldFinder
+        String result = EffectiveTldFinder.getAssignedDomain(hostname, true);
+        if (result == null) {
+        	LOGGER.debug(String.format("Hostname %s isn't a valid FQDN", hostname));
```

Comment:
You don't need to use StringFormat with slf4j: just do 
`("Hostname {} isn't a valid FQDN", hostname)`
also the call to String.format is causing the build to fail, see https://travis-ci.org/crawler-commons/crawler-commons/builds/293871975#L1048

Reply:
Hi @jnioche - Yeah, I've got projects with log4j and slf4j, I keep forgetting to use the slf4j formatting support. Also thanks for the heads-up re forbidden apis, I didn't realize I needed to run `mvn forbiddenapis:check` to catch these problems - I though `mvn package` did that.

Reply:
Fixed

Reply:
Hi @kkrugler, I had forgotten about forbiddenapis but that shows that the automated checks with PRs are very useful. I'll have a closer look at the PR on Tuesday. Are you happy with me releasing 0.9 before merging it? we could release 0.9.1 shortly after that

Reply:
Yes, please push out 0.9 - this is a very low priority change, just something I noticed in some code using the PLD support.